koans/src/java7/AboutTryWithResources.java

```diff
@@ -0,0 +1,106 @@
+package java7;
+
+import com.sandwich.koan.Koan;
+
+import java.io.*;
+
+import static com.sandwich.koan.constant.KoanConstants.__;
+import static com.sandwich.util.Assert.assertEquals;
+
+public class AboutTryWithResources {
+
+    @Koan
+    public void lookMaNoClose() {
+        String str = "first line"
+                + System.lineSeparator()
+                + "second line";
+        InputStream is = new ByteArrayInputStream(str.getBytes());
+        String line;
+        /* BufferedReader implementing @see java.lang.AutoCloseable interface */
+        try (BufferedReader br =
+                     new BufferedReader(
+                             new InputStreamReader(is))) {
+            line = br.readLine();
+            //br guaranteed to be closed
```

Comment:
I wonder if there is a way to assert this fact as part of the koan.  Unfortunately, there isn't an `isClosed` method on the `InputStream`.  Do you think it would be valuable to add a Koan that uses an implementation of `AutoCloseable` that implements its own `isClosed` method?